src/main/java/org/jitsi/videobridge/transform/RtpChannelTransformEngine.java

```diff
@@ -67,16 +67,28 @@
      * Initializes a new <tt>RtpChannelTransformEngine</tt> for a specific
      * <tt>RtpChannel</tt>.
      * @param channel the <tt>RtpChannel</tt>.
+     * @param manipulators video chain manipulators
      */
-    public RtpChannelTransformEngine(RtpChannel channel)
+    public RtpChannelTransformEngine(
+            RtpChannel channel,
+            Iterable<VideoTransformChainManipulator> manipulators)
     {
         this.channel = channel;
         this.logger
             = Logger.getLogger(
                     classLogger,
                     channel.getContent().getConference().getLogger());
 
-        engineChain = createChain();
+        TransformEngine[] chain = createChain();
+
+        if (manipulators != null)
+        {
+            for (VideoTransformChainManipulator manipulator : manipulators)
+            {
+                chain = manipulator.manipulate(chain);
```

Comment:
If I'm reading this correctly, it looks like the manipulator can't distinguish between a video and an audio stream. Would it make sense to pass the stream (this) as an argument to the manipulate method?

Reply:
Yes, the manipulator works with rtp chains, I will give the channel to manipulate function, so it can determinate the chain (channel) type. 

e.g.:
```boolean video = (channel instanceof VideoChannel);```